src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -706,6 +709,67 @@ public static IResourceBuilder<T> WithHealthCheck<T>(this IResourceBuilder<T> bu
         return builder;
     }
 
+    /// <summary>
+    /// Adds a health check to the resource which is mapped to a specific endpoint.
+    /// </summary>
+    /// <typeparam name="T">A resource type that implements <see cref="IResourceWithEndpoints" />.</typeparam>
+    /// <param name="builder">A resource builder.</param>
+    /// <param name="path">The relative path to test.</param>
+    /// <param name="statusCode">The result code to interpret as healthy.</param>
+    /// <param name="endpointName">The name of the endpoint to derive the base address from.</param>
+    /// <returns>A resource builder.</returns>
+    public static IResourceBuilder<T> WithHttpHealthCheck<T>(this IResourceBuilder<T> builder, string? path = null, int? statusCode = null, string? endpointName = null) where T : IResourceWithEndpoints
+    {
+        path = path ?? "/";
+        endpointName = endpointName ?? "http";
+        statusCode = statusCode ?? 200;
+
+        Uri? uri = null;
+        builder.ApplicationBuilder.Eventing.Subscribe<BeforeResourceStartedEvent>(builder.Resource, (@event, ct) =>
+        {
+            var endpoint = builder.Resource.GetEndpoint(endpointName);
+            var uriString = $"{endpoint.AllocatedEndpoint.UriString}{path}";
+            uri = new Uri(uriString, UriKind.Absolute);
```

Comment:
You don't want to mess with slashes right?

```suggestion
            uri = new Uri(endpoint.Url, path, UriKind.Absolute);
```